### DISCUSSION TITLE
Any way to use useSignMessage without manual connect by user?

### DISCUSSION THREAD

**Discussion by cryptoKevinL** [rainbow-me/rainbowkit#1302] [Q&A] [answered]
Hi, 

We want to prompt the user to sign a message with `useSignMessage`, without having the user manually connect.

Our use case is that we already know the users wallet address from their saved account settings in a different dApp, and even if they have not connected to our site before we want to prompt a given wallet address to sign our SIWE message.

Thanks!

**Comment by DanielSinclair** [comment, marked as answer]
This unfortunately won't be possible because wallet provider connection requests are account blind, and the account information is only exposed after a user approved a connection from within their wallet. For the dApp to communicate with a wallet to request a signature, you will also always need a connection established. A better way to handle this would be to ask the user to switch accounts in their wallet if the address returned from `useAccount()` does not match the expected address. Only when the address is a match would you kick off `useSignMessage`. Generally, though, you should expect users to own more than 1 address, and should really be thinking about 1-to-many user accounts.
